### DISCUSSION TITLE
What is the benefit of useMutation?

### DISCUSSION THREAD

**Discussion by devhyunjae** [TanStack/query#2447] [Q&A] [answered]
What is the benefit of useMutation comparing to the classic react-way?

example.

useMutation
```
useMutation(() => addMovie(movie, user?.id), {
    onSuccess: () => {
      queryClient.refetchQueries('recommendations')
    }
  })
```

vs

plain react
```
async () => {
   await addMovie(movie, user?.id)
   await queryClient.refetchQueries('recommendations')
}
```

**Comment by TkDodo** [comment, marked as answer]
You can track the state of the mutation, like loading and error. This allows you to e.g. disable the button that triggers the mutation while the mutation is running - a very common scenario

**Reply by devhyunjae** [reply]
Thanks for the fast response!
